src/main/java/org/springframework/integration/dsl/context/IntegrationFlowContext.java

```diff
@@ -149,8 +149,11 @@ public void register(String flowId, IntegrationFlow integrationFlow, boolean aut
 	 */
 	public synchronized void remove(String flowId) {
 		if (this.registry.containsKey(flowId)) {
+			IntegrationFlow theFlow = this.registry.remove(flowId);
```

Comment:
But don't we have the same problem there? An `IntegrationFlow` that is based on the adapter (or lambda) won't be started/stopped if registered dynamically.